### PR TITLE
Clarify logging for "No Project" when hitting memory limit

### LINKED ISSUE #59856: Clarify logging for "No Project" when hitting memory limit
### Acknowledgement

- [X] I acknowledge that issues using this template may be closed without further explanation at the maintainer's discretion.

### Comment

typescript fails with "No Project" and no other error in its log when hitting default memory limit. Improve this by providing better logging output.

### PR DESCRIPTION
**Pull request by cbarber**
typescript fails with "No Project" and no other error in its log when hitting default memory limit. Improve this by providing better logging output.

Fixes #59856

### CODE REVIEW

File: src/server/project.ts [hunk 1]
```diff
@@ -1071,6 +1071,8 @@ export abstract class Project implements LanguageServiceHost, ModuleResolutionHo
         this.clearGeneratedFileWatch();
         this.projectService.verifyDocumentRegistry();
         this.projectService.onUpdateLanguageServiceStateForProject(this, /*languageServiceEnabled*/ false);
+
+        this.error("Project service was disabled for exceeding memory limit. See disableSizeLimit config option.");
```
Comment (sheetalkamat) [inline src/server/project.ts:1075]: This is already logged. And shouldnt be logged as error.

File: src/server/session.ts [hunk 2]
```diff
@@ -2089,7 +2089,12 @@ export class Session<TMessage = string> implements EventSender {
         projects = filter(projects, p => p.languageServiceEnabled && !p.isOrphan());
         if (!ignoreNoProjectError && (!projects || !projects.length) && !symLinkedProjects) {
             this.projectService.logErrorForScriptInfoNotFound(args.file ?? args.projectFileName);
-            return Errors.ThrowNoProject();
+            if (some(projects, p => !p.languageServiceEnabled)) {
+                Errors.ThrowProjectLanguageServiceDisabled();
```
Comment (sheetalkamat) [inline src/server/session.ts:2093]: Can you show case where this happens. I dont think we ever check language service disabled when getting project

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
This is incorrect change

File: src/server/project.ts (hunk 1 above)
Comment (cbarber) [reply src/server/project.ts:1075]: Where is it logged?

Comment (sheetalkamat) [reply src/server/project.ts:1075]: getFilenameForExceededTotalSizeLimitForNonTsFiles

**Pull request closed without merging**